### PR TITLE
Refactor RetrievePetition for AmendPetition Scenarios. Removed return…

### PR DESCRIPTION
**Pull request by tchow8**
…ed AmendPetition cases for AOS users

# Description

[DIV-4227](https://tools.hmcts.net/jira/browse/DIV-4277)

This PR changes the algorithm for retrieving a case by the Petitioner.
The new flow is outlined in the ticket. Essentially:
1. Get both Draft and Cases for a User
2. Any case for the user will be returned in the following priority:
2.1 If multiple complete cases, return the latest one. (Excluding Rejected and AmendPetition)
2.2. If multiple incomplete cases (AwaitingPayment), throw DuplicateCaseFoundException.
2.3. Any non-Rejected and non-AmendPetition case.
2.4 Return latest AmendPetition case if exists.
3. Return non-Rejected and non-AmendPetition case if any found.
4. Otherwise if AmendPetition case exists:
4.1 Return amendPetition draft.
4.2 Return AmendPetition case in draft format
5. Otherwise return draft
6. Or if no draft, return empty response details.

This PR also removes the AmendPetition case state from being pulled by the Respondent retrieval code. This means that as soon as the Petitioner begins to amend a case, the Respondent will go back to the Capture Case Number and PIN page on login.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Unit and Integration tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tchow8** [comment]
Hi @alexwakeman @qzhou-hmcts, I've made a slight change to the logic, if you could quickly re-review please.

File: ruleset.xml [hunk 1]
```diff
@@ -6,26 +6,22 @@
...
   </rule>
-  <rule ref="rulesets/java/typeresolution.xml"/>
-  <rule ref="rulesets/java/typeresolution.xml/SignatureDeclareThrowsException">
+  <rule ref="category/java/design.xml/SignatureDeclareThrowsException">
     <properties>
       <property name="IgnoreJUnitCompletely" value="true"/>
     </properties>
   </rule>
-  <rule ref="rulesets/java/unnecessary.xml"/>
-  <rule ref="rulesets/java/unusedcode.xml"/>
```
Comment (qzhou-hmcts) [inline ruleset.xml]: ???

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/CcdRetrieveCaseTest.java [hunk 2]
```diff
@@ -168,9 +172,63 @@ public void givenDoNotCheckCcdAndMultiplePetitionsInDivorceFormat_whenRetrieveCa
...
+    @Test
+    public void givenAmendPetitionCaseAndNoDraft_whenRetrieveCase_thenReturnAmendDraft() throws Exception {
+        final UserDetails userDetails = getUserDetails();
+
+        createACaseMakePaymentAndAmendTheCase(userDetails.getAuthToken());
+
+        Response cmsResponse = retrieveCase(userDetails.getAuthToken(), true);
+
+        assertEquals("true", cmsResponse.getBody().jsonPath().getString("case_data.fetchedDraft"));
+        assertEquals(null, cmsResponse.getBody().jsonPath().getString("case_data.previousCaseId"));
```
Comment (qzhou-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/CcdRetrieveCaseTest.java]: this is wrong, when you retrieve the amendDraft previousCaseId has to be populated.

**Comment by qzhou-hmcts** [comment]
@tchow8  this is not part of your PR but I just realise this line that worries me https://github.com/hmcts/div-case-maintenance-service/blob/90834911167dfa4b6f5875486cc5fde031a04398/src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java#L162

@alexwakeman do you know why are we setting the divorceUnit when we are generating the amend petition?

File: ruleset.xml (hunk 1 above)
Comment (tchow8) [reply ruleset.xml]: Upgraded to PMD 6.11 and removed deprecated rulesets, which have since been combined into the other categories.

**Comment by alexwakeman** [comment]
@qzhou-hmcts we assign the divorce unit as per the ACs of part 6 of https://tools.hmcts.net/jira/browse/DIV-3744

**Pull request merged**